addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1268,81 +1271,15 @@ options.registry.OptionsTab = options.Class.extend({
         await this.customizeWebsiteVariable(previewMode, this.bodyImageType, {variable: 'body-image-type'});
         await this.customizeWebsiteVariable(previewMode, widgetValue ? `'${widgetValue}'` : '', {variable: 'body-image'});
     },
+
     /**
+     * Directly hook OptionTab::openCustomCodeDialog to website.editor.custom_code::openCustomCodeDialog
      * @see this.selectClass for parameters
      */
     async openCustomCodeDialog(previewMode, widgetValue, params) {
-        const libsProm = loadBundle({
-            jsLibs: [
-                '/web/static/lib/ace/ace.js',
-                '/web/static/lib/ace/mode-xml.js',
-                '/web/static/lib/ace/mode-qweb.js',
-            ],
-        });
-
-        let websiteId;
-        this.trigger_up('context_get', {
-            callback: (ctx) => {
-                websiteId = ctx['website_id'];
-            },
-        });
-
-        let website;
-        const dataProm = this._rpc({
-            model: 'website',
-            method: 'read',
-            args: [[websiteId], ['custom_code_head', 'custom_code_footer']],
-        }).then(websites => {
-            website = websites[0];
-        });
-
-        let fieldName, title, contentText;
-        if (widgetValue === 'head') {
-            fieldName = 'custom_code_head';
-            title = _t('Custom head code');
-            contentText = _t('Enter code that will be added into the <head> of every page of your site.');
-        } else {
-            fieldName = 'custom_code_footer';
-            title = _t('Custom end of body code');
-            contentText = _t('Enter code that will be added before the </body> of every page of your site.');
-        }
-
-        await Promise.all([libsProm, dataProm]);
+        return await openHeadBodyEditor(this)
```

Comment:
`return await` almost never makes sense, you can just `return` I think here :+1: